tests/unit/parts/test_lifecycle.py

```diff
@@ -1351,6 +1354,64 @@ def test_lifecycle_run_in_provider_try(
     )
 
 
+@pytest.mark.parametrize(
+    "snapcraft_base, provider_base", SNAPCRAFT_BASE_TO_PROVIDER_BASE.items()
+)
+def test_lifecycle_run_in_provider_bases(
+    emitter,
+    snapcraft_base,
+    provider_base,
+    mock_get_instance_name,
+    mock_instance,
+    mock_provider,
+    mocker,
+    snapcraft_yaml,
+    tmp_path,
+):
+    """Verify snapcraft bases are handled properly when launching an instance."""
+    mocker.patch(
+        "snapcraft.projects.Project.get_effective_base", return_value=snapcraft_base
+    )
+    mock_base_configuration = Mock()
+    mocker.patch(
+        "snapcraft.parts.lifecycle.providers.get_base_configuration",
+        return_value=mock_base_configuration,
+    )
+    mocker.patch("snapcraft.parts.lifecycle.providers.capture_logs_from_instance")
+    mocker.patch("snapcraft.parts.lifecycle.providers.ensure_provider_is_available")
+    mocker.patch("snapcraft.parts.lifecycle.providers.prepare_instance")
+    mocker.patch("snapcraft.projects.Project.get_build_on")
+    mocker.patch("snapcraft.projects.Project.get_build_for")
+
+    project = Project.unmarshal(snapcraft_yaml(base="core22"))
+    parts_lifecycle._run_in_provider(
+        project=project,
+        command_name="test",
+        parsed_args=argparse.Namespace(
+            use_lxd=False,
+            debug=False,
+            bind_ssh=False,
+            http_proxy=None,
+            https_proxy=None,
+        ),
+    )
+
+    mock_provider.launched_environment.assert_called_with(
+        project_name="mytest",
+        project_path=ANY,
+        base_configuration=mock_base_configuration,
+        build_base=provider_base.value,
+        instance_name="test-instance-name",
+        allow_unstable=bool(snapcraft_base == "devel"),
+    )
+
+    if snapcraft_base == "devel":
```

Comment:
It's very inconvenient, but this should probably be its own test that runs the lifecycle just for `devel` and checks that it output this.

Reply:
Done!  I split the test into 2 separate unit tests.  Let me know if the tests are as discrete as you imagined.

Reply:
Very nice!